CONTRIBUTING.md

```diff
@@ -1,77 +1,150 @@
+
+### [WIP] Code Contribution Process
+Content will be added to this section once the reposority is ready for open source contribution. 
+
+### Contribution Bar
+The WinUI team accepts code changes that improve WinUI or fix bugs, as long as they follow the processes outlined below and broadly align with our [roadmap](docs/roadmap.md).
+
+While we strive to accept all community contributions that meet the guidelines outlined here, please note that we may not merge changes that have narrowly-defined benefits due to compatibility risks and maintenance costs. We may also revert changes if they are found to be breaking.
+
+### Contributor License Agreement
 Most contributions require you to agree to a Contributor License Agreement (CLA) declaring that you have the right to, and actually do, grant us the rights to use your contribution. For details, visit https://cla.microsoft.com.
 
 When you submit a pull request, a CLA-bot will automatically determine whether you need to provide a CLA and decorate the PR appropriately (e.g., label, comment). Simply follow the instructions provided by the bot. You will only need to do this once across all repos using our CLA.
 
 ### Copying files from other projects
-
 The following rules must be followed for PRs that include files from another project:
+- The license of the file is [permissive](https://en.wikipedia.org/wiki/Permissive_free_software_licence).
+- The license of the file is left intact.
+- The contribution is correctly attributed in the [3rd party notices](https://github.com/dotnet/coreclr/blob/master/THIRD-PARTY-NOTICES.TXT) file in the repository, as needed.
+
+## Commit Messages
+Please format commit messages as follows (based on [A Note About Git Commit Messages](http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html)):
+
+```
+Summarize change in 50 characters or less
+
+Provide more detail after the first line. Leave one blank line below the 
+summary and wrap all lines at 72 characters or less.
 
-* The license of the file is [permissive](https://en.wikipedia.org/wiki/Permissive_free_software_licence).
-* The license of the file is left intact.
-* The contribution is correctly attributed in the [3rd party notices](https://github.com/dotnet/coreclr/blob/master/THIRD-PARTY-NOTICES.TXT)
-file in the repository, as needed.
+If the change fixes an issue, leave another blank line after the final 
+paragraph and indicate which issue is fixed in the specific format below.
 
-## Documentation and usage samples
+Fix #42
+```
 
-You can also read and contribute to the WinUI documentation here:  
-https://docs.microsoft.com/uwp/toolkits/winui
+## Checks
+Each pull request to main must pass checks within Azure DevOps. This pipelines build your change and run automated tests. These tests should match what you're able to run with local automated testing using Test Explorer. They also create a NuGet package to match your change.
```

Comment:
> local automated testing using Test Explorer

Recommend removing "using Test Explorer", since most tests are unlikely to be run via Test Explorer at least initially.